### PR TITLE
Add minikube prow testing docker image

### PR DESCRIPTION
**Pull request by azhao155**
Add minikube prow testing docker image, has verified as following:

```
bazel run //prow/cmd/mkpj -- --config-path=/home/zyanshu/test-infra/config/prow/config.yaml --job-config-path=/home/zyanshu/test-infra/config/jobs/kubernetes-sigs/kind/minikube-local.yaml --job=pull-minikube-make > /tmp/foo
bazel run //prow/cmd/phaino -- /tmp/foo --privileged
```

and prow testing succeeded.
```
* Preparing Kubernetes v1.20.2 on Docker 20.10.6 ...
  - Generating certificates and keys ...                                                                                                                                             - Booting up control plane ...
  - Configuring RBAC rules ...
* Verifying Kubernetes components...                                                                                                                                                 - Using image gcr.io/k8s-minikube/storage-provisioner:v5
* Enabled addons: storage-provisioner, default-storageclass
* Done! kubectl is now configured to use "minikube" cluster and "default" namespace by default                                                                                     NAMESPACE     NAME                               READY   STATUS    RESTARTS   AGE
kube-system   etcd-minikube                      0/1     Running   0          10s
kube-system   kube-apiserver-minikube            1/1     Running   0          10s                                                                                                  kube-system   kube-controller-manager-minikube   0/1     Running   0          10s
kube-system   kube-scheduler-minikube            0/1     Running   0          10s
kube-system   storage-provisioner                0/1     Pending   0          2s                                                                                                   wrapper.sh] [TEST] Test Command exit code: 0
wrapper.sh] [CLEANUP] Cleaning up after Docker in Docker ...                                                                                                                       a8fff0cb45cb
Stopping Docker: dockerwrapper.sh] [CLEANUP] Done cleaning up after Docker in Docker.
================================================================================                                                                                                   wrapper.sh] Exiting 0
.
INFO[0308] PASS: e2e84320-ab21-11eb-ba5a-42010a800002    duration=5m8.551682628s job=pull-minikube-make                                                                            INFO[0308] SUCCESS
INFO[0308] Press Ctrl + c to exit.
^CINFO[0738] Received signal.                              signal=interrupt
INFO[0738] Interrupt received.                                                                                                                                                     INFO[0738] All workers gracefully terminated, exiting.
```

### CODE REVIEW

File: deploy/prow/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        && mkdir -p "${GOPATH}/bin" \
+    && echo "Installing Docker ..." \
+        && curl -fsSL https://download.docker.com/linux/$(. /etc/os-release; echo "$ID")/gpg | apt-key add - \
+        && add-apt-repository \
+            "deb [arch=amd64] https://download.docker.com/linux/$(. /etc/os-release; echo "$ID") \
+            $(lsb_release -cs) stable" \
+        && apt-get update \
+        && apt-get install -y --no-install-recommends docker-ce \
+        && rm -rf /var/lib/apt/lists/* \
+        && sed -i 's/cgroupfs_mount$/#cgroupfs_mount\n/' /etc/init.d/docker \
```
Comment (medyagh) [inline deploy/prow/Dockerfile:74]: Add comment or issue link

Comment (azhao155) [reply deploy/prow/Dockerfile:74]: has comment "# TODO: the `sed` is a bit of a hack, look into alternatives." before Installing Packages

File: Makefile [hunk 2]
```diff
@@ -100,6 +100,9 @@ GVISOR_TAG ?= latest
 # auto-pause-hook tag to push changes to
 AUTOPAUSE_HOOK_TAG ?= 1.13
 
+# prow-test tag to push changes to
+PROW_TEST_TAG ?= 1.7
```
Comment (medyagh) [inline Makefile:104]: lets change this to a sem ver format, how about v0.0.1 ?

**Review by medyagh — CHANGES_REQUESTED** [review]

File: Makefile (hunk 2 above)
Comment (azhao155) [reply Makefile]: Done

**Review by medyagh — APPROVED** [review]

**Comment by medyagh** [comment]
Great work, thank you very much @azhao155

This work took many prototypes and many tries. I admire your persistence and great work

**Pull request merged**